src/collectors/redisstat/redisstat.py

```diff
@@ -268,9 +261,42 @@ def _get_info(self, host, port, auth):
         # Pull full stats. Use execute_command for backwards compatibility
         # with older redis clients.
         info = client.execute_command('INFO', 'all')
-        del client
         return info
 
+    def _reset_stats(self, host, port, auth):
+        """Resets statistics on the specified Redis instance
+
+:param str host: redis host
+:param int port: redis port
+:rtype: dict
+
+        """
+        client = self._client(host, port, auth)
+        if client is None:
+            return None
+
+        # Reset statistics. Use execute_command for backwards compatibility
+        # with older redis clients.
+        client.execute_command('CONFIG RESETSTAT')
```

Comment:
I'd rather not go down the road of having any meta-monitoring logic like that in Diamond (I'm not aware of it existing for any other collector already).

I intentionally did not have an inner try/except for this - it won't actually publish any metrics if it can't reset statistics, so the worst case is everything goes to zero, which in theory would trigger alarms on its own.